modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -198,7 +279,36 @@ public Error Resize(int newSize)
         }
 
         /// <summary>
-        /// Shuffles the contents of this <see cref="Array"/> into a random order.
+        /// Reverses the order of the elements in the array.
+        /// </summary>
+        public void Reverse()
+        {
+            var self = (godot_array)NativeValue;
+            NativeFuncs.godotsharp_array_reverse(ref self);
+        }
+
+        /// <summary>
+        /// Compares this <see cref="Array"/> against the <paramref name="other"/>
+        /// <see cref="Array"/> by value.
+        /// </summary>
+        /// <param name="other">The other array to compare against.</param>
+        /// <returns>
+        /// <see langword="true"/> if the sizes and contents of the arrays are equal,
+        /// <see langword="false"/> otherwise.
+        /// </returns>
+        public bool SequenceEqual(Array other)
```

Comment:
In C# a method of this name would normally compare elements by reference and not recursively by value, the docs should be clearer that the behavior differs from the other `SequenceEqual` methods.

Reply:
The `Enumerable.SequenceEqual` method uses the default equality comparer (if none is specified) which will use `IEquatable` if implemented by the element's type (it's usually recommended to do so for Value Types and I believe all our variant types implement it).
It is my understanding that that's how the `recursive_equal` method works too when recursion count is 0.

Reply:
```csharp
> new Godot.Collections.Array() { 0 }.Equals(new Godot.Collections.Array() { 0 })
false
> new [] { 0 }.Equals(new [] { 0 })
false
> new Godot.Collections.Array() { new Godot.Collections.Array() { 0 } }.SequenceEqual(new Godot.Collections.Array() { new Godot.Collections.Array() { 0 } })
true
> new[] { new[] { 0 } }.SequenceEqual(new [] { new[] { 0 } })
false
```
`recursion_count` is the number of recursions already done, not the number to do.

Reply:
Ah, I see, you are right. In that case I would prefer to avoid the name `SequenceEqual`. I'll rename it to `RecursiveEqual`.